solutions/swb-reference/src/services/keyPairService.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  CreateKeyPairRequest,
+  CreateKeyPairResponse,
+  DeleteKeyPairRequest,
+  GetKeyPairRequest,
+  GetKeyPairResponse,
+  KeyPair,
+  KeyPairParser,
+  KeyPairPlugin,
+  ListKeyPairsRequest,
+  SendPublicKeyRequest,
+  SendPublicKeyResponse
+} from '@aws/swb-app';
+import { DynamoDBService, PaginatedResponse } from '@aws/workbench-core-base';
+
+export default class KeyPairService implements KeyPairPlugin {
+  private _dynamoDbService: DynamoDBService;
+  private _resourceType: string = 'key-pair';
+  public constructor(dynamoDbService: DynamoDBService) {
+    this._dynamoDbService = dynamoDbService;
+  }
+
+  /**
+   * Gets a Key Pair record
+   *
+   * @param request - a {@link GetKeyPairRequest}
+   * @returns a {@link KeyPair}
+   */
+  public getKeyPair(request: GetKeyPairRequest): Promise<GetKeyPairResponse> {
+    // TODO implement
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Lists Key Pair records
+   *
+   * @param request - a {@link ListKeyPairsRequest}
+   * @returns a {@link PaginatedResponse} of {@link KeyPair}s
+   */
+  public listKeyPairs(request: ListKeyPairsRequest): Promise<PaginatedResponse<KeyPair>> {
+    // TODO implement
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Deletes a Key Pair record
+   *
+   * @param request - a {@link DeleteKeyPairRequest}
+   */
+  public deleteKeyPair(request: DeleteKeyPairRequest): Promise<void> {
+    // TODO implement
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Creates a Key Pair record
+   *
+   * @param request - a {@link CreateKeyPairRequest}
+   * @returns a {@link CreateKeyPairResponse}
+   */
+  public createKeyPair(request: CreateKeyPairRequest): Promise<CreateKeyPairResponse> {
+    // TODO implement
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * Sends the public key of a Key Pair to the designated environment to allow connection
+   *
+   * @param request - a {@link SendPublicKeyRequest}
+   */
+  public sendPublicKey(request: SendPublicKeyRequest): Promise<SendPublicKeyResponse> {
+    // TODO implement
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * This method formats a Key Pair object as a DDB item containing key pair data
+   *
+   * @param keyPair - the {@link KeyPair} object to prepare for DDB
+   * @returns an object containing key pair data as well as pertinent DDB attributes based on key pair data
+   */
+  private _mapToDDBItemFromKeyPair(keyPair: KeyPair): Record<string, string> {
```

Comment:
I know we've done this type of transformation previously, but converting to a `Record<string, string>` means we're losing information about what is actually going into dynamo db. What do you think of having a custom class for DynamoDB objects similar to whats happening in `_mapDDBItemToKeyPair`?

Reply:
Open to it! I'll sync with you!